pageScraper.js

```diff
@@ -1,91 +1,48 @@
-  return await pagePromise(url1, url2, waitSelector, isbn, browser)
-}
-
+import { urls } from "./siteResources.js";
+const isbns = ["0061964360", "0199535566"];
+// const isbns = ['9781599664026']
 
-async function pagePromise(url1, url2, waitSelector, isbn, browser) {
-  const url = url2 ? (url1 + isbn + url2) : (url1 + isbn);
-  const page = await browser.newPage();
+async function pagePromise(url, fields, page) {
   console.log(`Navigating to ${url}.`);
-  await page.goto(url);
-  // Wait for the required DOM elements to be loaded
-  try {
-    // See if item can be found on Booklog
-    await page.waitForSelector(waitSelector, {timeout: 2500} );
-  } 
-  catch (e) {
-    console.log("Book not found at booklog.jp.\n Trying bookfinder.com...");
-    return false
-   }
-
-
-  // Find DOM elements and store found text in object
-  let dataObj = {};
-
-  dataObj["Title"] = await page.$eval('[itemprop=name]', text => text.textContent) 
-  dataObj["Author"] = await page.$eval('[itemprop=author]' , text => text.textContent) 
-  dataObj["Publisher"] = await page.$eval('[itemprop=publisher]', text => text.textContent)
-  let about = await page.$eval('[itemprop=description]', text => text.textContent)
-  let pageCount = await page.$eval('.info-area > ul > li', text => text.textContent)
-
-  dataObj["Page Count"] = pageCount.split("(")[1].slice(0,3);
-  dataObj["About"] = about;
-
-  return dataObj
-}
-
-const timeGenerator = () => {
-  // Random number generator used for random timeout values
-  let num = Math.round((Math.random(3) * 4) * 1000) 
-  return (num > 1000) ? num : 1231
-}
-
-function timeout(ms) {
-  // Timeout function so the bot doesn't query too many times too fast
-  return new Promise(resolve => setTimeout(resolve, ms));
+  await page.goto(url, {
+    waitUntil: "networkidle2",
+  });
+  const result = Object.entries(fields).reduce(
+    async (acc, [field, queryString]) => {
+      try {
+        const value = await page.$eval(queryString, (text) => text.textContent);
+        return { ...await acc, [field]: value };
+      } catch(err) {
+        console.log('Failed pluck for:', field);
+        return acc;
+      }
+    },
+  {});
+
+  console.warn('Reduced value:', JSON.stringify(result));
+
+  return result;
 }
 
-export default async function scraper(browser) {
-  // Go through isbns in list -> search on site
-  for (let i = 0; i < isbns.length; i++) {
-    const time = timeGenerator()
-    let currentPageData = await booklogQuery(isbns[i], browser);
-    if (currentPageData) {
-      // If the first url returns false (waitforSelector times out), try next url
-      console.log(currentPageData) // will be replaced with data push
-      await timeout(time)
-      console.log("Time waited before execution", time)
-    } else {
-      currentPageData = await bookfinderQuery(isbns[i], browser);
-      console.log(currentPageData) // will be replaced with data push
+const iterateThroughURLs = (urls, page) => async (data, isbn) => {
```

Comment:
Higher order function (function that returns a function) that is being used by the reduce in `scraper`. 

We go through each of the sites from the urls config until we get the data or we return an empty object.